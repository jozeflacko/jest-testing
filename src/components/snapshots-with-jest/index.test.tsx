import * as React from 'react';
import * as renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<a className="link" href="http://www.google.com">Google</a>).toJSON();
  expect(tree).toMatchSnapshot();
});

/*
// using prettier
it('renders correctly with inline snashotu', () => {
    const tree = renderer.create(<a className="link" href="http://www.google.com">Google</a>).toJSON();
    expect(tree).toMatchInlineSnapshot();
});*/

/**
 * Property Matchers
 */
/*
it('will check the matchers and pass', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'LeBron James',
    };
    //will fail every time
    //expect(user).toMatchSnapshot();

    
    expect(user).toMatchSnapshot<typeof user>({
        createdAt: expect.any(Date),
        id: expect.any(Number)
    });
  });
  */