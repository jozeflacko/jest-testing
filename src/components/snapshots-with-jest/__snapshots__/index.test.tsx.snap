// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="link"
  href="http://www.google.com"
>
  Google
</a>
`;
